astropy_timeseries/sampled.py

```diff
@@ -161,22 +171,38 @@ def from_pandas(self, df):
         # TODO: determine how user can specify time scale
         time = Time(df.index)
 
-        # Create table without the time column
-        table = Table.from_pandas(df)
+        if LooseVersion(astropy.__version__) > LooseVersion("3.1.1"):
+            table = Table.from_pandas(df, index=index)
+        else:
+            table = Table.from_pandas(df)
 
         return TimeSeries(time=time, data=table)
 
-    def to_pandas(self):
+    def to_pandas(self, index=None):
         """
-        Convert this time series to a :class:`~pandas.DataFrame` with a
-        :class:`~pandas.DatetimeIndex` index.
+        Convert this :class:`~astropy_timeseries.TimeSeries` to a
+        :class:`~pandas.DataFrame` with a :class:`~pandas.DatetimeIndex` index.
+
+        Parameters
+        ----------
+        index : None, bool, str
+            Specify DataFrame index mode
+            Only used if Astropy version >=``3.2``.
```

Comment:
Can you clarify this docstring? (the one in ``from_pandas`` is clearer, and it's not clear what passing a ``str`` would do here)

Reply:
I assume you would like some tests for the new index feature?

Reply:
Yes please, but this doesn't have to be too extensive.